Feature unification conflict because of `encoding` feature

I'm building an app which is split into multiple crates. To simplify the setup, let's assume the following:
- Crate A depends on `quick-xml` with the `encoding` feature enabled
- Crate B depends on `quick-xml` with the `encoding` feature _**not**_ enabled and uses the `Attribute::unescape_value` function
- Crate C depends on crates A & B

When trying to build crate C, I get the following error message:
```
error[E0599]: no method named `unescape_value` found for struct `Attribute` in the current scope
  --> B\src\lib.rs:9:7
   |
6  | /     Attribute {
7  | |         key: QName(b"key"),
8  | |         value: Cow::Borrowed(b"value"),
9  | |     }.unescape_value()
   | |______-^^^^^^^^^^^^^^
   |
help: there is a method `decode_and_unescape_value` with a similar name, but with different arguments
  --> C:\Users\Maxence\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\quick-xml-0.37.5\src\events\attributes.rs:79:5
   |
79 |     pub fn decode_and_unescape_value(&self, decoder: Decoder) -> XmlResult<Cow<'a, str>> {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

It is clear that `B` expects `quick-xml` to not have the `encoding` feature enabled.

Is there a way to solve this feature unification conflict?
Here's a repo to reproduce the issue: https://github.com/maxence-cornaton/quick-xml-feature-unification

---
For a real world scenario, imagine having crate A depending on [calamine](https://github.com/tafia/calamine) and crate B depending on [mail_auth](https://docs.rs/mail-auth/latest/mail_auth/). Crate will fail to compile because `calamine` enables the `encoding` feature while `mail_auth` expects it to be disabled.
```
error[E0599]: no method named `unescape_value` found for struct `Attribute` in the current scope
   --> C:\Users\Maxence\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\mail-auth-0.6.1\src\report\dmarc\parse.rs:312:48
    |
312 |                         if let Ok(attr) = attr.unescape_value() {
    |                                                ^^^^^^^^^^^^^^
    |
help: there is a method `decode_and_unescape_value` with a similar name, but with different arguments
   --> C:\Users\Maxence\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\quick-xml-0.37.5\src\events\attributes.rs:79:5
    |
79  |     pub fn decode_and_unescape_value(&self, decoder: Decoder) -> XmlResult<Cow<'a, str>> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

--- a/src/events/attributes.rs
+++ b/src/events/attributes.rs
@@ -41,9 +41,18 @@ impl<'a> Attribute<'a> {
     ///
     /// See also [`unescape_value_with()`](Self::unescape_value_with)
     ///
-    /// This method is available only if [`encoding`] feature is **not** enabled.
+    /// <div style="background:rgba(120,145,255,0.45);padding:0.75em;">
+    ///
+    /// NOTE: Because this method is available only if [`encoding`] feature is **not** enabled,
+    /// should only be used by applications.
+    /// Libs should use [`decode_and_unescape_value()`](Self::decode_and_unescape_value)
+    /// instead, because if lib will be used in a project which depends on quick_xml with
+    /// [`encoding`] feature enabled, the lib will fail to compile due to [feature unification].
+    ///
+    /// </div>
     ///
     /// [`encoding`]: ../../index.html#encoding
+    /// [feature unification]: https://doc.rust-lang.org/cargo/reference/features.html#feature-unification
     #[cfg(any(doc, not(feature = "encoding")))]
     pub fn unescape_value(&self) -> XmlResult<Cow<'a, str>> {
         self.unescape_value_with(resolve_predefined_entity)
@@ -60,9 +69,18 @@ impl<'a> Attribute<'a> {
     ///
     /// See also [`unescape_value()`](Self::unescape_value)
     ///
-    /// This method is available only if [`encoding`] feature is **not** enabled.
+    /// <div style="background:rgba(120,145,255,0.45);padding:0.75em;">
+    ///
+    /// NOTE: Because this method is available only if [`encoding`] feature is **not** enabled,
+    /// should only be used by applications.
+    /// Libs should use [`decode_and_unescape_value_with()`](Self::decode_and_unescape_value_with)
+    /// instead, because if lib will be used in a project which depends on quick_xml with
+    /// [`encoding`] feature enabled, the lib will fail to compile due to [feature unification].
+    ///
+    /// </div>
     ///
     /// [`encoding`]: ../../index.html#encoding
+    /// [feature unification]: https://doc.rust-lang.org/cargo/reference/features.html#feature-unification
     #[cfg(any(doc, not(feature = "encoding")))]
     #[inline]
     pub fn unescape_value_with<'entity>(

